trollflow2/tests/test_launcher.py

```diff
@@ -887,5 +887,69 @@ def test_generate_messages():
             assert msg.type in VALID_MESSAGE_TYPES
 
 
+def test_sigterm_generate_messages():
+    """Test that sending sigterm to Trollflow2 stops it."""
+    import os
+    import signal
+    from multiprocessing import Process
+
+    from trollflow2.launcher import generate_messages
+
+    connection_parameters = {"nameserver": False, "addresses": "localhost:40000", "topic": "/test"}
+    proc = Process(target=generate_messages, args=(connection_parameters, ))
+    proc.start()
+    # Wait for the message listening loop to start
+    time.sleep(1)
+    # Send SIGTERM
+    os.kill(proc.pid, signal.SIGTERM)
+    proc.join()
+
+    assert proc.exitcode == 0
+
+
+def _fake_queue_logged_process(msg, prod_list, produced_files, **kwargs):
+    time.sleep(5.0)
+
+
+@mock.patch("trollflow2.launcher.ListenerContainer")
```

Comment:
Do we really need this mocking? would it work with posstroll.testing's `patched_subscriber_recv`? 
example usage: https://github.com/pytroll/pytroll-runner/blob/main/pytroll_runner/tests/test_runner.py#L323

Reply:
How does it work replacing `ListenerContainer`, the underlying `Listener` and the output queue?

Reply:
I tried this:
```python
    with patched_subscriber_recv([msg]):
        runner = Runner(product_list, connection_parameters)
    proc = Process(target=runner.run)
    proc.start()
    ...
```

but it exited immediately with exitcode 15 (SIGTERM) after the `kill` command.

Reply:
So `_fake_queue_logged_process`isn't called?

Reply:
Yes, seems so, and the test completes in 1.007 seconds.

Reply:
Ok, thinking more about this, I would pass two messages to the patched subscriber and check that only one is processed.
But that doesn't help with the fake process being ignored. The patching is still there, right?
I don't understand why this is not working, the patching of the subscriber only passes messages down. Unless the ListenerContainer bypasses it somehow? Can you check if the message is consumed?

Reply:
Ok, I think I figured it out. Things broke at https://github.com/pytroll/posttroll/blob/main/posttroll/listener.py#L47 because I had empty connection settings...

Switched to `patched_subscriber_recv` in https://github.com/pytroll/trollflow2/pull/217/commits/aefd2a33fd131f24065dcbafd123d968680c2406

But the `fake_queue_logged_process` is still not run. If I change it to sleep for 15 s the execution time doesn't change. But lets continue with that outside this thread.